datacapturing/src/main/java/de/cyface/datacapturing/DataCapturingService.java

```diff
@@ -947,11 +952,38 @@ public void onBindingDied(final @NonNull ComponentName name) {
         }
     }
 
+    /**
+     * Adds a new {@link DataCapturingListener} interested in events from the {@link DataCapturingBackgroundService}.
+     * <p>
+     * All listeners are automatically removed when the {@link DataCapturingService} is killed.
+     *
+     * @param listener A listener that is notified of important events during data capturing.
+     * @return true if this collection changed as a result of the call
+     */
+    @SuppressWarnings("unused") // Used by SDK implementing apps (S, C)
+    public boolean addDataCapturingListener(@NonNull final DataCapturingListener listener) {
+        return fromServiceMessageHandler.addListener(listener);
+    }
+
+    /**
+     * Removes a registered {@link DataCapturingListener} from {@link DataCapturingBackgroundService} events.
+     * <p>
+     * Listeners may be removed when on Android's onPause Lifecycle method or e.g. when the UI is disabled.
```

Comment:
What happens if I remove a listener while the background service is running? Maybe unit test this.

Reply:
nothing should happen. The currently released 1.5.1214 already does this when the app is paused during capturing. I added a test anyways.